js/src/lib/ajax-filter.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @package Polylang
+ */
+
+/**
+ * Adds data to all ajax requests made with jQuery.
+ *
+ * @since 3.7
+ *
+ * @param {Object} data The data to add.
+ * @returns {void}
+ */
+const ajaxFilter = ( data ) => {
+	if ( 'undefined' === typeof jQuery ) {
+		return;
+	}
+
+	jQuery(
+		function ( $ ) {
+			const dataStr = $.param( data );
+
+			$.ajaxPrefilter( function ( options ) {
+				if ( -1 === options.url.indexOf( ajaxurl ) && -1 === ajaxurl.indexOf( options.url ) ) {
+					return;
+				}
+
+				if (
+					'undefined' === typeof options.data ||
+					null === options.data ||
+					'string' === typeof options.data && '' === options.data.trim()
+				) {
+					// An empty string or null/undefined.
+					options.data = dataStr;
+				} else if ( 'string' === typeof options.data ) {
+					// A non-empty string: can be a JSON string or a query string.
+					try {
+						options.data = JSON.stringify( Object.assign( JSON.parse( options.data ), data ) );
+					} catch ( exception ) {
+						// A non-empty non-JSON string is considered a query string.
+						options.data = `${ options.data }&${ dataStr }`;
+					}
+				} else if ( $.isPlainObject( options.data ) ) {
+					// An object.
+					options.data = Object.assign( options.data, data );
+				}
+			} );
+		}
+	);
+}
+
+export default ajaxFilter;
```

Comment:
I tend to avoid default export nowadays, that we can always import using `{}` and don't need to think wether or not it is a default one.
Also, if some you want to add a new function, it doable more easily.
But I let you see what you prefer.

Reply:
Makes sense.
Done in [105b1a8](https://github.com/polylang/polylang/pull/1612/commits/105b1a80bbeb122b7588a4e2031f8b022ef9dea2).